packages/suite/src/utils/wallet/tokenUtils.ts

```diff
@@ -90,16 +90,41 @@ export const getTokens = (
 
         if (searchQuery && !isTokenMatchesSearch(token, query)) return;
 
+        const hasBalance = token.balance !== '0';
+
         if (isShown) {
```

Comment:
how about something like this, but not strong opinion 

```
const hasBalance = token.balance !== '0';
const pushToArray = (arrayWithBalance, arrayWithoutBalance) => {
    if (hasBalance) {
        arrayWithBalance.push(token);
    } else {
        arrayWithoutBalance.push(token);
    }
};

if (isShown) {
    pushToArray(shownWithBalance, shownWithoutBalance);
} else if (hasCoinDefinitions && !isKnown) {
    pushToArray(unverifiedWithBalance, unverifiedWithoutBalance);
} else if (isHidden) {
    pushToArray(hiddenWithBalance, hiddenWithoutBalance);
} else {
    pushToArray(shownWithBalance, shownWithoutBalance);
}
```

Reply:
Looks cleaner 👍